Microsoft.Toolkit.Uwp/Helpers/HttpHelper.cs

```diff
@@ -57,13 +68,49 @@ protected HttpHelper()
         {
             var httpRequestMessage = request.ToHttpRequestMessage();
 
-            var response = await _httpClient.SendRequestAsync(httpRequestMessage).AsTask().ConfigureAwait(false);
+            var client = await GetHttpClientInstance();
+
+            var response = await client.SendRequestAsync(httpRequestMessage).AsTask().ConfigureAwait(false);
+
+            // Add the HttpClient instance back to the queue.
+            _httpClientQueue.Enqueue(client);
 
             FixInvalidCharset(response);
 
             return new HttpHelperResponse(response);
         }
 
+        private async Task<HttpClient> GetHttpClientInstance()
+        {
+            HttpClient client;
+
+            // Try and get HttpClient from the queue
+            if (!_httpClientQueue.TryDequeue(out client))
+            {
+                if (_queueCount == MaxPoolSize)
+                {
+                    do
+                    {
+                        // HttpClient connection queue all instances in use.. Wait
+                        resetEvent.Wait(50);
```

Comment:
resetEvent has to be set/unset
like set it when releasing one httpClient
and just wait with resetEvent.WaitOne()

The idea is just to get a non active wait. And not pooling either (even every 50s).

And after thinking about it we should even think about using a semaphoreSlim(MaxPoolSize) here. This should be even better